eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -9,53 +9,185 @@ import {
+  for (const model of resourceModels) {
+    const metadata = resourceModelMap.get(model.crossLanguageDefinitionId);
+    if (metadata) {
+      addResourceMetadata(model, metadata);
+    }
+  }
+}
 
-  // the last step, add the decorator to the client
-  for (const client of clients) {
-    const resourceMetadata = metadata.get(client);
-    if (resourceMetadata) {
-      addResourceMetadata(client, resourceMetadata);
+function parseResourceOperation(
+  serviceMethod: SdkMethod<SdkHttpOperation> | undefined,
+  sdkContext: SdkContext<AzureEmitterOptions, SdkHttpOperation>
+): [ResourceOperationKind, string | undefined] | undefined {
+  const decorators = serviceMethod?.__raw?.decorators;
+  for (const decorator of decorators ?? []) {
+    if (decorator.definition?.name === armResourceReadName) {
+      return [
+        ResourceOperationKind.Get,
+        getResourceModelId(sdkContext, decorator)
+      ];
+    } else if (decorator.definition?.name == armResourceCreateOrUpdateName) {
+      return [
+        ResourceOperationKind.Create,
+        getResourceModelId(sdkContext, decorator)
+      ];
+    } else if (decorator.definition?.name == armResourceUpdateName) {
+      return [
+        ResourceOperationKind.Update,
+        getResourceModelId(sdkContext, decorator)
+      ];
+    } else if (decorator.definition?.name == armResourceDeleteName) {
+      return [
+        ResourceOperationKind.Delete,
+        getResourceModelId(sdkContext, decorator)
+      ];
+    } else if (decorator.definition?.name == armResourceListName) {
+      return [
+        ResourceOperationKind.List,
+        getResourceModelId(sdkContext, decorator)
+      ];
+    } else if (decorator.definition?.name == armResourceActionName) {
+      return [
+        ResourceOperationKind.Action,
+        getResourceModelId(sdkContext, decorator)
+      ];
     }
   }
+  return undefined;
 }
 
-export function getAllClients(codeModel: CodeModel): InputClient[] {
-  const clients: InputClient[] = [];
-  for (const client of codeModel.clients) {
+function getParentResourceModelId(
+  sdkContext: SdkContext<AzureEmitterOptions, SdkHttpOperation>,
+  model: SdkModelType | undefined
+): string | undefined {
+  const decorators = (model?.__raw as Model)?.decorators;
+  const parentResourceDecorator = decorators?.find(
+    (d) => d.definition?.name == parentResourceName
+  );
+  return getResourceModelId(sdkContext, parentResourceDecorator) ?? undefined;
+}
+
+function getResourceModelId(
+  sdkContext: SdkContext<AzureEmitterOptions, SdkHttpOperation>,
+  decorator?: DecoratorApplication
+): string | undefined {
+  if (!decorator) return undefined;
+  const model = getClientType(
+    sdkContext,
+    decorator.args[0].value as Model
+  ) as SdkModelType;
+  if (model) {
+    return model.crossLanguageDefinitionId;
+  } else {
+    throw new Error(
+      `Resource model not found for decorator ${decorator.decorator.name}`
+    );
```

Comment:
I think we should report error diagnostic here.
It could be `general-error` for now.